code-generators/java-types/buildSrc/src/main/java/uk/gov/di/codegen/CustomObjectRule.java

```diff
@@ -0,0 +1,42 @@
+package uk.gov.di.codegen;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.sun.codemodel.JDefinedClass;
+import com.sun.codemodel.JPackage;
+import com.sun.codemodel.JType;
+import org.jsonschema2pojo.Schema;
+import org.jsonschema2pojo.rules.ObjectRule;
+import org.jsonschema2pojo.rules.RuleFactory;
+import org.jsonschema2pojo.util.ParcelableHelper;
+import org.jsonschema2pojo.util.ReflectionHelper;
+
+import java.util.Map;
+
+import static org.jsonschema2pojo.util.TypeUtil.resolveType;
+
+/**
+ * Custom object rule that includes interfaces for classes with a given suffix
+ */
+public class CustomObjectRule extends ObjectRule {
+    private static final Map<String, String> INTERFACE_MAPPING = Map.of(
+            "CredentialJWTClass", "uk.gov.di.model.CredentialJWTClass",
+            "CredentialClass", "uk.gov.di.model.CredentialClass"
```

Comment:
This is obviously not a scalable approach - it fixes the most fundamental problem of not having a shared interface for the different credential types (which is by far the most problematic), but:
- Those interfaces don't contain any content, so you still have to cast it to a specific type of credential to do anything with it
- It doesn't cover other relationships - e.g. it would be nice to capture the fact that `IdentityCheck extends PersonWithIdentity`, or `Passport extends Document`

Solving this fully is likely impossible with the current approach, as json-schema doesn't capture these relationships. 😞